src/Nethermind/Nethermind.TxPool/LightTransaction.cs

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using Nethermind.Core;
+
+namespace Nethermind.TxPool;
+
+/// <summary>
+/// For sorting reasons - without storing full, large txs in memory
+/// </summary>
+public class LightTransaction : Transaction
```

Comment:
I'm not happy about removing it. It can be useful in the future. Right now it is derived from Transaction object to be compatible with all old `Transaction` collections, comparers etc. In the future we can potentially optimize TxPool by refactoring it to use only this object for sorting reasons (and drop link with Transaction - it will be separate, light object) and keep raw tx RLPs in separate collection, without regular gas bottleneck updates (so removals and adding back). We will be able to use such light collection when building blocks and load full txs only when are picked to the block. I'm not sure if it will improve our performance enough to be worth it, but integrating it with Transaction even closer will make this experiment harder to perform